### PR TITLE
[plugin] support for more clients

### PR DESCRIPTION
**Pull request by yance-dev**
Resurrecting master
## What changes are proposed in this pull request?

subscription was created by web js clinet as a user token :

https://github.com/voxel51/fiftyone/blob/a0b2956172bef30af9f50ca418b6fac3805fe691/app/packages/looker/src/index.ts#:~:text=const%20subscription%20%3D%20uuid()%3B

we use it to bulid subscription specific state pool.

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: fiftyone/server/events.py [hunk 1]
```diff
@@ -55,11 +56,13 @@ async def dispatch_event(
 
     events = []
     for listener in _listeners[event.get_event_name()]:
+        break
```
Comment (phoenix-xhuang) [inline fiftyone/server/events.py]: 直接在56行上面return怎么样？

**Comment by phoenix-xhuang** [comment]
如果pr准备好了，可以在reviewer里面指定一下，这样我们会收到通知，能及时review

File: fiftyone/server/events.py [hunk 2]
```diff
@@ -183,24 +186,29 @@ async def dispatch_polling_event_listener(
     }
 
 
-def get_state() -> fos.StateDescription:
+def get_state(subscription) -> fos.StateDescription:
     """Get the current state description singleton on the server
```
Comment (fenrir-z) [inline fiftyone/server/events.py]: 这个说明是否也要改一下？

File: fiftyone/server/events.py [hunk 3]
```diff
@@ -183,24 +186,29 @@ async def dispatch_polling_event_listener(
...
-    return _state
 
 
 async def _disconnect(
     is_app: bool, listeners: t.Set[t.Tuple[str, Listener]]
 ) -> None:
     for event_name, listener in listeners:
         _listeners[event_name].remove(listener)
+        # remove state
+    del _state_pool[listener.subscription]
```
Comment (fenrir-z) [inline fiftyone/server/events.py]: 是否需要先判定 `listener.subscription` 一定在 `dict` 中存在？

File: fiftyone/server/events.py (hunk 1 above)
Comment (yance-dev) [reply fiftyone/server/events.py]: 已处理

File: fiftyone/server/events.py (hunk 3 above)
Comment (yance-dev) [reply fiftyone/server/events.py]: 已增加处理逻辑

File: fiftyone/server/events.py (hunk 2 above)
Comment (yance-dev) [reply fiftyone/server/events.py]: 已更新注释

**Review by phoenix-xhuang — APPROVED** [review]

**Review by fenrir-z — APPROVED** [review]

**Pull request merged**